R/zchunk_LA119.solar.R

```diff
@@ -29,51 +28,56 @@ module_gcam.usa_LA119.Solar_DISABLED <- function(command, ...) {
     # Load required inputs
     states_subregions <- get_data(all_data, "gcam-usa/states_subregions")
     NREL_us_re_capacity_factors <- get_data(all_data, "gcam-usa/NREL_us_re_capacity_factors")
-    NREL_us_re_technical_potential <- get_data(all_data, "gcam-usa/NREL_us_re_technical_potential")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` and/or 'match' calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Merge-and-Match
-    # ===================================================
+    # Perform computations: Create scalars to scale capacity factors read in in the assumptions file in the energy folder to adjust for varying irradiance by state.
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
+    NREL_us_re_capacity_factors %>%
+      gather(fuel, average, -State) %>%
+      filter (State == "Average") %>%
+      select (-State) ->
+      Average_fuel
+
+    NREL_us_re_capacity_factors %>%
+      gather(fuel, value, -State) %>%
+      filter (State != "Average") %>%
+      left_join(Average_fuel, by = "fuel") %>%
+      mutate(scaler = value / average, sector = "electricity generation") %>%
+      select(State, sector, fuel, scaler) ->
+      NREL_us_re_capacity_factors
+
+    states_subregions %>%
+      mutate(State = state_name) %>%
+      select(state, State) %>%
+      left_join(NREL_us_re_capacity_factors, by = "State") %>%
+      select(-State) ->
+      NREL_us_re_capacity_factors
+
+    NREL_us_re_capacity_factors %>%
+      filter(fuel == "Urban_Utility_scale_PV") %>%
+      mutate(fuel = "solar PV") ->
+      L119.CapFacScaler_PV_state
+
+    NREL_us_re_capacity_factors %>%
+      filter(fuel == "CSP") %>%
+      mutate(fuel = "solar CSP") ->
+      L119.CapFacScaler_CSP_state
+
+    L119.CapFacScaler_PV_state %>%
       add_title("descriptive title of data") %>%
       add_units("units") %>%
       add_comments("comments describing how data generated") %>%
       add_comments("can be multiple lines") %>%
       add_legacy_name("L119.CapFacScaler_PV_state") %>%
```

Comment:
The above lines, i.e. `title`, `units`, and `comments` need to be filled out with descriptive and accurate information.